diff --git a/toolchain-m68k b/toolchain-m68k
index 566781e..f82beea 100755
--- a/toolchain-m68k
+++ b/toolchain-m68k
@@ -22,6 +22,7 @@ URLS = \
    'ftp://ftp.gnu.org/gnu/binutils/binutils-2.9.1.tar.gz',
    'ftp://ftp.gnu.org/gnu/gcc/gcc-2.95.3/gcc-core-2.95.3.tar.gz',
    'ftp://ftp.gnu.org/gnu/gcc/gcc-2.95.3/gcc-g++-2.95.3.tar.gz',
+   'ftp://ftp.gnu.org/gnu/ncurses/ncurses-5.9.tar.gz',
    ('http://fossies.org/linux/misc/old/flex-2.5.4a.tar.gz',
     'flex-2.5.4.tar.gz'),
    'git://github.com/cahirwpz/libnix',
@@ -197,6 +198,12 @@ def build():
   environ['LANG'] = 'C'
   environ['TERM'] = 'xterm'
 
+  find_executable('perl')
+  find_executable('patch')
+  find_executable('make')
+  find_executable('git')
+  find_executable('yacc')
+
   with cwd('{archives}'):
     for url in URLS:
       if type(url) == tuple:
@@ -205,6 +212,9 @@ def build():
         name = path.basename(url)
       fetch(name, url)
 
+  unpack('python-lha', work_dir='{build}')
+  python_setup('python-lha')
+
   add_site_dir('{host}')
 
   """
@@ -236,25 +246,20 @@ def build():
 
   """
   When we have a working compiler in our path, we shoule also check if the
-  required programs, headers and libraries are present.
+  required headers and libraries are present.
   """
 
-  find_executable('perl')
-  find_executable('patch')
-  find_executable('make')
-  find_executable('git')
-  find_executable('yacc')
-
-  require_header('ncurses.h', 'c', 'libncurses5-dev package missing',
-                 'NCURSES_VERSION_MAJOR', 5)
-
-  py_ver = 'python%d.%d' % (sys.version_info.major, sys.version_info.minor)
-
-  require_header(path.join(py_ver, 'Python.h'), 'c',
-                 'python-dev package missing')
-
-  unpack('python-lha', work_dir='{build}')
-  python_setup('python-lha')
+  unpack('{ncurses}')
+  configure('{ncurses}',
+            '--prefix={host}',
+            '--without-debug',
+            '--without-cxx-binding',
+            '--without-tests',
+            '--without-ada',
+            '--without-manpages',
+            '--without-shared')
+  make('{ncurses}')
+  make('{ncurses}', 'install')
 
   unpack('{m4}')
   configure('{m4}', '--prefix={host}')
@@ -589,7 +594,8 @@ if __name__ == "__main__":
          binutils_ver=args.binutils,
          gcc_ver=args.gcc)
 
-  setvar(m4='m4-1.4.17',
+  setvar(ncurses='ncurses-5.9',
+         m4='m4-1.4.17',
          gawk='gawk-3.1.8',
          flex='flex-2.5.4',
          bison='bison-1.35',
